commcare_connect/program/forms.py

```diff
@@ -48,3 +58,21 @@ def save(self, commit=True):
             instance.created_by = self.user.email
         instance.modified_by = self.user.email
         return super().save(commit=commit)
+
+
+class ManagedOpportunityInitForm(OpportunityInitForm):
+    class Meta(OpportunityInitForm.Meta):
+        model = ManagedOpportunity
+
+    def __init__(self, *args, **kwargs):
+        self.program = kwargs.pop("program")
+        super().__init__(*args, **kwargs)
+
+        # Managed opportunities should use the currency specified in the program.
+        self.fields["currency"].initial = self.program.currency
+        self.fields["currency"].widget = forms.TextInput(attrs={"readonly": "readonly", "disabled": True})
+        self.fields["currency"].required = False
+
+    def save(self, commit=True):
+        self.instance.program = self.program
```

Comment:
We should also add a statement to set the Organization to None as the ManagedOpportunity comes under a Program. This Organization field can later be used for Network Managers.
Does this sound correct to you? @calellowitz 
```suggestion
        self.instance.program = self.program
        self.instance.organization = None
```

Reply:
yeah that sounds right, and is the exact reason we made the field nullable